### PR TITLE
Added Support for PutUsbScanCodes on each VM

### PR DESCRIPTION
**Pull request by king-jam**
This adds this capability to govc: https://www.virtuallyghetto.com/2017/09/automating-vm-keystrokes-using-the-vsphere-api-powercli.html

### CODE REVIEW

File: govc/vm/putusbscancodes.go [hunk 1]
```diff
@@ -0,0 +1,316 @@
...
+func (cmd *putusbscancode) Usage() string {
+	return ""
+}
+
+func (cmd *putusbscancode) Description() string {
+	return `Send Keystroke to VM.
+
+Examples:
+ Default Scenario
+  govc vm.putusbscancode $vm -r 1376263 (writes an 'r' to the console)
```
Comment (dougm) [inline govc/vm/putusbscancodes.go]: Tried it out, the examples should be `-vm $vm`.  Or to support both, can do this (and add "VM" to Usage): https://github.com/vmware/govmomi/blob/cf359b62d1e6a76bcfeee0d5cc1bfb8f7db685b7/govc/vm/console.go#L82-L91

File: govc/vm/putusbscancodes.go [hunk 2]
```diff
@@ -0,0 +1,316 @@
...
+  govc vm.putusbscancode $vm -c 0x15 (writes an 'r' to the console)
```
Comment (dougm) [inline govc/vm/putusbscancodes.go]: Can you make these a comment instead of parens `# writes an 'r' to the console`

**Comment by king-jam** [comment]
@dougm Updated per your feedback.

I didn't end up using the console.go example because I encountered an issue with the VM Search function reporting `no arguments` when I provided the `-vm` flag or if I defined  `GOVC_VM`. Bug?

**Comment by yusufozturk** [comment]
Hi all,

First of all, thanks for this extension in the API. @king-jam @dougm 

Our automation process uses PutUsbScanCodes to wake up virtual machines so we can capture their console screens and put it into the AI engine so we can process if there is any anomality on the virtual machine screen.

Everything works great except everytime when we run PutUsbScanCodes, it creates an event on vCenter. There are thousand of virtual machines and we do this process every 5 minutes due to screening and compliance checks so you can image the mess on the vcenter event console.

image

Is there any way to supress the event? If there is no way to supress this event, do you think we can do this via another way?

Thanks,

Yusuf

**Pull request merged**